images/base/files/usr/local/bin/entrypoint

```diff
@@ -101,22 +101,30 @@ overlayfs_preferrable() {
 
 configure_containerd() {
   local snapshotter=${KIND_EXPERIMENTAL_CONTAINERD_SNAPSHOTTER:-}
-  if [[ -n "$userns" ]]; then
-    # userns (rootless) configs
 
-    # Adjust oomScoreAdj
+  # handle userns (rootless)
+  if [[ -n "$userns" ]]; then
+    # enable restrict_oom_score_adj
     sed -i 's/restrict_oom_score_adj = false/restrict_oom_score_adj = true/' /etc/containerd/config.toml
-
     # Use fuse-overlayfs if overlayfs is not preferrable: https://github.com/kubernetes-sigs/kind/issues/2275
     if [[ -z "$snapshotter" ]] && ! overlayfs_preferrable; then
       snapshotter="fuse-overlayfs"
     fi
-  else
+  fi
+
+  # if we have not already overridden the snapshotter, attempt to auto select
+  if [[ -z "$snapshotter" ]]; then
     # we need to switch to the 'native' snapshotter on zfs
-    if [[ -z "$snapshotter" ]] && [[ "$(stat -f -c %T /kind)" == 'zfs' ]]; then
+    container_filesystem="$(stat -f -c %T /kind)"
+    if [[ "$container_filesystem" == 'zfs' ]]; then
       snapshotter="native"
```

Comment:
Not new in this PR but I guess we can use fuse-overlayfs on zfs

Reply:
Hmm true, I think if we did we'd have to also check that fuse is actually present to not regress on https://github.com/kubernetes-sigs/kind/issues/2540

Reply:
done.